template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/oozie/OozieRoleConfigProvider.java

```diff
@@ -13,36 +14,39 @@
 import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
 import com.sequenceiq.cloudbreak.api.endpoint.v4.database.base.DatabaseType;
 import com.sequenceiq.cloudbreak.cmtemplate.configproviders.AbstractRdsRoleConfigProvider;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils;
 import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
 import com.sequenceiq.cloudbreak.template.views.HostgroupView;
 import com.sequenceiq.cloudbreak.template.views.RdsView;
 
 @Component
 public class OozieRoleConfigProvider extends AbstractRdsRoleConfigProvider {
 
-    private static final String OOZIE_DATABASE_HOST = "oozie_database_host";
-
-    private static final String OOZIE_DATABASE_NAME = "oozie_database_name";
-
     private static final String OOZIE_DATABASE_TYPE = "oozie_database_type";
 
     private static final String OOZIE_DATABASE_USER = "oozie_database_user";
 
     private static final String OOZIE_DATABASE_PASSWORD = "oozie_database_password";
 
+    private static final String OOZIE_DATABASE_JDBC_URL = "oozie.service.JPAService.jdbc.url";
+
     private static final String OOZIE_CONFIG_SAFETY_VALVE = "oozie_config_safety_valve";
 
     @Override
     protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
         switch (roleType) {
             case OozieRoles.OOZIE_SERVER:
                 RdsView oozieRdsView = getRdsView(source);
-                List<ApiClusterTemplateConfig> config = new ArrayList<>(List.of(
-                    config(OOZIE_DATABASE_HOST, oozieRdsView.getHost()),
-                    config(OOZIE_DATABASE_NAME, oozieRdsView.getDatabaseName()),
-                    config(OOZIE_DATABASE_TYPE, oozieRdsView.getSubprotocol()),
-                    config(OOZIE_DATABASE_USER, oozieRdsView.getConnectionUserName()),
-                    config(OOZIE_DATABASE_PASSWORD, oozieRdsView.getConnectionPassword())));
+                StringBuilder ozzieSafetyVale = new StringBuilder();
+                ozzieSafetyVale.append(ConfigUtils
+                        .getSafetyValveProperty(OOZIE_DATABASE_JDBC_URL, oozieRdsView.getConnectionURL()));
+                List<ApiClusterTemplateConfig> config = new ArrayList<>();
+                config.add(config(OOZIE_DATABASE_TYPE, oozieRdsView.getSubprotocol()));
+                config.add(config(OOZIE_DATABASE_USER, oozieRdsView.getConnectionUserName()));
+                config.add(config(OOZIE_DATABASE_PASSWORD, oozieRdsView.getConnectionPassword()));
+                if (!ozzieSafetyVale.toString().isEmpty()) {
+                    config.add(config(OOZIE_CONFIG_SAFETY_VALVE, ozzieSafetyVale.toString()));
+                }
```

Comment:
I see your point, but I wonder if the safety valve can ever by empty here. If it was indeed possible, `OOZIE_DATABASE_HOST` and `OOZIE_DATABASE_NAME` would be missing from `config`, so either we need an `else` branch, or remove the `if` around the safety valve altogether.